third_party/rapidjson/CMakeLists.txt

```diff
@@ -1,10 +1,14 @@
-message(STATUS "Configuring and building rapidjson")
-ginkgo_load_git_package(rapidjson_external
-    "https://github.com/Tencent/rapidjson.git"
-    "88bd956d66d348f478bceebfdadb8e26c6844695"
-    "-DRAPIDJSON_BUILD_DOC=OFF" "-DRAPIDJSON_BUILD_EXAMPLES=OFF"
-    "-DRAPIDJSON_BUILD_TESTS=OFF" "-DRAPIDJSON_BUILD_THIRDPARTY_GTEST=OFF"
-    "-DRAPIDJSON_BUILD_CXX11=ON")
-# Remove directory with questionable licensing
-file(REMOVE_RECURSE ${CMAKE_CURRENT_BINARY_DIR}/src/bin/jsonchecker)
-ginkgo_add_external_target(rapidjson rapidjson src/include "" STATIC "" rapidjson_external TRUE)
+message(STATUS "Fetching external RapidJSON")
+include(FetchContent)
+FetchContent_Declare(
+    rapidjson
+    GIT_REPOSITORY https://github.com/Tencent/rapidjson.git
+    GIT_TAG        88bd956d66d348f478bceebfdadb8e26c6844695
+)
+FetchContent_GetProperties(rapidjson)
+if(NOT rapidjson_POPULATED)
+    FetchContent_Populate(rapidjson)
+endif()
+set(RapidJSON_INCLUDE_DIR "${rapidjson_SOURCE_DIR}/include")
+add_library(rapidjson INTERFACE)
```

Comment:
doesn't rapidjson provide their target?

Reply:
Unfortunately no, they only provide a target in their RapidJSONConfig.cmake, which is only available in installed RapidJSON, but not if we include it in our build.